tests/Unit/ApparentHorizons/Test_ApparentHorizonFinder.cpp

```diff
@@ -321,69 +359,81 @@ void test_apparent_horizon(const gsl::not_null<size_t*> test_horizon_called,
-        get<gr::Solutions::KerrSchild::DerivShift<DataVector>>(solution_vars);
-    const auto& dt_shift =
-        get<Tags::dt<gr::Tags::Shift<3, Frame::Inertial, DataVector>>>(
-            solution_vars);
-    const auto& g =
-        get<gr::Tags::SpatialMetric<3, Frame::Inertial, DataVector>>(
-            solution_vars);
-    const auto& dt_g =
-        get<Tags::dt<gr::Tags::SpatialMetric<3, Frame::Inertial, DataVector>>>(
-            solution_vars);
-    const auto& d_g =
-        get<gr::Solutions::KerrSchild::DerivSpatialMetric<DataVector>>(
-            solution_vars);
-
-    // Fill output variables with solution.
-    typename ::Tags::Variables<typename metavars::interpolator_source_vars>::
-        type output_vars(mesh.number_of_grid_points());
-    get<::gr::Tags::SpacetimeMetric<3, Frame::Inertial>>(output_vars) =
-        gr::spacetime_metric(lapse, shift, g);
-    get<::GeneralizedHarmonic::Tags::Phi<3, Frame::Inertial>>(output_vars) =
-        GeneralizedHarmonic::phi(lapse, d_lapse, shift, d_shift, g, d_g);
-    get<::GeneralizedHarmonic::Tags::Pi<3, Frame::Inertial>>(output_vars) =
-        GeneralizedHarmonic::pi(
-            lapse, dt_lapse, shift, dt_shift, g, dt_g,
-            get<::GeneralizedHarmonic::Tags::Phi<3, Frame::Inertial>>(
-                output_vars));
-
-    // Call the InterpolatorReceiveVolumeData action on each element_id.
-    ActionTesting::simple_action<interp_component,
-                                 intrp::Actions::InterpolatorReceiveVolumeData>(
-        make_not_null(&runner), mock_core_for_each_element.at(element_id),
-        first_temporal_id, element_id, mesh, output_vars);
-    ActionTesting::simple_action<interp_component,
-                                 intrp::Actions::InterpolatorReceiveVolumeData>(
-        make_not_null(&runner), mock_core_for_each_element.at(element_id),
-        second_temporal_id, element_id, mesh, std::move(output_vars));
+                            typename metavars::AhA>>(make_not_null(&runner), 0,
+                                                     temporal_ids);
+
+  // Create volume data and send it to the interpolator, for each temporal_id.
+  for (const auto& temporal_id: temporal_ids) {
+    for (const auto& element_id : element_ids) {
+      const auto& block = domain.blocks()[element_id.block_id()];
+      ::Mesh<3> mesh{domain_creator->initial_extents()[element_id.block_id()],
+                     Spectral::Basis::Legendre,
+                     Spectral::Quadrature::GaussLobatto};
+
+      tnsr::I<DataVector, 3, Frame::Inertial> inertial_coords{};
+      if constexpr (IsTimeDependent::value) {
+        ElementMap<3, Frame::Grid> map_logical_to_grid{
+            element_id, block.moving_mesh_logical_to_grid_map().get_clone()};
+        // We don't have an Element ParallelComponent in this test, so
+        // get the cache from the target component.
+        auto& cache = ActionTesting::cache<target_component>(runner, 0_st);
```

Comment:
(Only if you need to make changes for CI failures) This can be `const auto&` I think